File: api/src/main/java/de/hipp/pnp/api/fivee/abstracts/BaseCharacterClass.java
package de.hipp.pnp.api.fivee.abstracts;


public abstract class BaseCharacterClass {

    private String name;
    private Integer level;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getLevel() {
        return level;
    }

    public void setLevel(Integer level) {
        this.level = level;
    }

    public void increaseLevel(int amount) {
        this.level = (this.level == null ? amount : this.level + amount);
    }
}

File: api/src/main/kotlin/de/hipp/pnp/api/fivee/abstracts/BaseCharacterClass.kt
package de.hipp.pnp.api.fivee.abstracts

abstract class BaseCharacterClass {
    open var name: String = ""
    open var level: Int? = 1

    fun increaseLevel(amount: Int) {
        this.level = (this.level?.plus(amount)) ?: amount
    }
}
